forum/qa-content/qa-page.js

```diff
@@ -206,73 +424,113 @@ function qa_ajax_error()
-			block.setAttribute('data-lang', languages[block.classList[0]]);
+		{									
+			var additionalDiv = document.createElement('div');
+			var blockButton = document.createElement('button');
+			var languageInfo = document.createElement('div');
+			var copyCodeBtn = document.createElement('button');
+							
+			additionalDiv.classList.add('syntaxhighlighter-additional');
 			
-			// when code-block has new lines and their number is greater than maximum number of lines before being collapsed
-			if (block.innerHTML.indexOf('\n') > -1 && block.innerHTML.match(/\n/g).length + 1 >= numberOfLines)
+			languageInfo.classList.add('syntaxhighlighter-language');
+							
+			// check how many lines of code each block has - if more than number in 'numberOfLines' variable, then make it collapsable
+			if (block.querySelectorAll('.line').length >= numberOfLines || ( block.innerHTML.indexOf('\n') > -1 && block.innerHTML.match(/\n/g).length + 1 >= numberOfLines ) )
 			{
+				blockButton.classList.add('syntaxhighlighter-button');
+				blockButton.textContent = '-- Rozwiń --';
+				
 				// add CSS class to do some styling
-				block.classList.add('collapsed');
-				// add attribute to let CSS :before pseudo element set it's content to appropriate state of code-block
-				block.setAttribute('data-state', '-- Rozwiń --');
-
+				block.classList.add('collapsed-block');
+				
 				// when user clicks on code-block
-				block.addEventListener('click', function(ev)
+				blockButton.addEventListener('click', function(ev)
 				{					
+					// prevent... dummy (refresh page) default action of button
+					ev.preventDefault();
+					
 					/*
-					 * when block-code is collapsed or not - change <pre> attribute and add/remove CSS class
-					 * to notify user the state of code-block
-					 */
-					if (block.classList.contains('collapsed'))
+					* when block-code is collapsed or not - write info on button and add/remove CSS class
+					* to notify user the state of code-block
+					*/
+					if (block.classList.contains('collapsed-block'))
 					{
-						block.classList.remove('collapsed');
-						block.setAttribute('data-state', '-- Zwiń --');
+						block.classList.remove('collapsed-block');
+						blockButton.textContent = '-- Zwiń --';
 					}
 					else
 					{
-						block.classList.add('collapsed');
-						block.setAttribute('data-state', '-- Rozwiń --');
+						block.classList.add('collapsed-block');
+						blockButton.textContent = '-- Rozwiń --';
 					}
 				});
+				
+				additionalDiv.appendChild(blockButton);
 			}
+			
+			// find out what language is used inside block
+			languageInfo.textContent = languages[block.classList[1]] || languages[block.classList[0].slice(block.classList[0].indexOf(':') + 1, -1)];
+			
+			additionalDiv.appendChild(languageInfo);
+			
+			copyCodeBtn.textContent = 'Kopiuj';
+			copyCodeBtn.classList.add('content-copy-btn');
+			
+			// when Browser supports Copying To Clipboard and SyntaxHighlighter is loaded

```

Comment:
In my opinion 

> // find out what language is used inside block
is useful comment, because it informs that below assignment searches for the name of language, which is used inside the code-block. Especially, that searching may not be clear to undestand for people who will look at that code in the future (I mean using `.classList`, `.slice` and `.indexOf` in one "combo" including `||` operator).

I changed it to "// based on each code-block CSS class - find out what language is used inside it"
However if You will still insist on removing that comment (because You think that this code may be enough readable for others), i do that.